presto-main/src/main/java/com/facebook/presto/sql/gen/NullIfCodeGenerator.java

```diff
@@ -101,7 +101,7 @@ private static BytecodeNode cast(
             Type actualType,
             Type requiredType)
     {
-        if (actualType.getTypeSignature().equals(requiredType)) {
+        if (actualType.equals(requiredType)) {
```

Comment:
I think this is safer.
```suggestion
        if (actualType.getTypeSignature().equals(requiredType.getTypeSignature())) {
```

Reply:
done